### PR TITLE
[MO] Move redundant checks in ScatterUpdate operation shape infer

### PR DESCRIPTION
**Pull request by achetver**
Root cause analysis: In case of dynamic input shapes `ScatterUpdate` operations changes and we cannou infer it using shape inference function from MO.

Solution: Move redundant checks in `ScatterUpdate` operation shape inference function (as we don't have such restrictions in specification), to operation `ScatterElementUpdate` where they should be done.

Ticket: 78217

Code:
* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR
* [x]  Transformation preserves original framework node names
* [x]  Transformation preserves tensor names

Validation:
* [x]  Unit tests: N/A - No new implemented functions;
* [x]  Framework operation tests: N/A - No new enabled operations;
* [x]  Transformation tests: N/A - No new implemented transformations;
* [x]  Model Optimizer IR Reader check: Done manually.

Documentation:
* [x]  Supported frameworks operations list: N/A - No new supported operations;
* [x]  Supported **public** models list: N/A - Nothing to add;
* [x]  User guide update: N/A - Nothing to update.

### CODE REVIEW

**Review by rkazants — CHANGES_REQUESTED** [review]
We have a bug in shape inference for Scatter operation. The current checks on validness of input shapes are incorrect based on the specification.
Please fix it there instead of creating extender because it relates to not only ONNX but can affect TF, etc. The fix shouldn't be difficult.

**Comment by achetver** [comment]
> We have a bug in shape inference for Scatter operation. The current checks on validness of input shapes are incorrect based on the specification. Please fix it there instead of creating extender because it relates to not only ONNX but can affect TF, etc. The fix shouldn't be difficult.

Ok. I removed these checks because we don't have such restrictions in op specification.

File: tools/mo/openvino/tools/mo/ops/scatter.py [hunk 1]
```diff
@@ -41,11 +41,6 @@ def infer(node: Node):
         updates_shape = node.in_port(2).data.get_shape()
         assert input_shape is not None and updates_shape is not None and indices_shape is not None, \
             'The node "{}" input shape is None'.format(node_name)
-        assert len(input_shape) == len(indices_shape), 'data and indices inputs for node "{}" must be of the ' \
-            'same rank. Instead got {} and {}'.format(node_name, len(input_shape), len(indices_shape))
-        assert compatible_shapes(indices_shape, updates_shape), \
-            'updates and indices shapes for node "{}" must be equal. Instead got {} and {}.' \
-            ''.format(node_name, indices_shape, updates_shape)
```
Comment (rkazants) [inline tools/mo/openvino/tools/mo/ops/scatter.py:48]: These checks should be done for `ScatterElementsUpdate`

Comment (achetver) [reply tools/mo/openvino/tools/mo/ops/scatter.py:48]: Thanks, done.

File: tools/mo/openvino/tools/mo/ops/scatter.py [hunk 2]
```diff
@@ -101,12 +96,23 @@ def infer(node: Node):
...
-                node_name, axis.size)
+            assert axis.size == 1, "The node {} has axis input value size equal to {} but it should be exactly 1.".\
+                format(node_name, axis.size)
+            assert len(input_shape) == len(indices_shape), 'data and indices inputs for node "{}" must be of the ' \
+                                                           'same rank. Instead got {} and {}'.format(node_name,
+                                                                                                     len(input_shape),
+                                                                                                     len(indices_shape))
+            assert compatible_shapes(indices_shape, updates_shape), \
+                'updates and indices shapes for node "{}" must be equal. Instead got {} and {}.' \
+                ''.format(node_name, indices_shape, updates_shape)
```
Comment (rkazants) [inline tools/mo/openvino/tools/mo/ops/scatter.py:115]: these checks should be on line 104

**Review by rkazants — APPROVED** [review]

**Pull request merged**